### ISSUE TITLE
spec(README): API in the README ?

### ISSUE THREAD

**Issue by douglasduteil** [douglasduteil/angular-utility-belt#7]
Will we put the full directive/service/provider/etc.. on the root README.md ?

**Comment by douglasduteil** [comment]
If yes how will we render the API of a directive/service/provider/etc.. in markdown ? Will it be like :
- https://github.com/gulpjs/gulp/blob/master/docs/API.md
- https://github.com/angular-ui/ui-router/wiki/Quick-Reference

**Comment by revolunet** [comment]
what about a table ?

**Comment by douglasduteil** [comment]
K 

So it will be like: 

---

### API

### <Directive name>

``` html
<ANY ng-model=""
       [name=""]
       [min=""]
       [max=""]>
```

| Param | Type | Details |
| --- | --- | --- |
| ngModel | `string` | Assignable angular expression to data-bind to. |
| name _(optional)_ | `string` | Property name of the form under which the control is published. |
| min _(optional)_ | `string` | Sets the <code>min</code> validation error key if the value entered is less than <code>min</code> |
| max _(optional)_ | `string` | Sets the <code>max</code> validation error key if the value entered is greater than <code>max</code>. |
| ... | ... | ... |

---

### $http

#### $http.get(url, [config]);

Shortcut method to perform GET request.

<table>
  <thead>
    <tr>
      <th>Param</th>
      <th>Type</th>
      <th>Details</th>
    </tr>
  </thead>
  <tbody>
     <tr>
      <td>
        ul
      </td>
      <td>
        <code>string</code>
      </td>
      <td>
        <p>Relative or absolute URL specifying the destination of the request</p>
        
      </td>
    </tr>
    <tr>
 <td>
        config
        <div><em>(optional)</em></div>
      </td>
      <td>
        <code>Object</code>
      </td>
      <td>
        <p>Object describing the request to be made and how it should be processed. The object has following properties: </p>
<ul>
<li><strong>method</strong> – <code>{string}</code> – HTTP method (e.g. 'GET', 'POST', etc)</li>
<li><strong>url</strong> – <code>{string}</code> – Absolute or relative URL of the resource that is being requested.</li>
<li>...</li>
</ul>
        
      </td>
     
    </tr>
    
  </tbody>
</table>

**Comment by revolunet** [comment]
yep je trouve ca tres bien !

**Comment by mgcrea** [comment]
Pareil, c'est top.

**Comment by mgcrea** [comment]
Initial support landed in 8e355a8ace9a7f384a2a21a827dae072bbaaa18c.
Looks like multiline could be a problem.

Check the [generated readme](https://github.com/douglasduteil/angular-utility-belt/blob/master/README.md)!
Related task (run in sequence before the readme's one): https://github.com/douglasduteil/angular-utility-belt/blob/master/ng-factory/tasks/docs/ngdocs.js

**Comment by douglasduteil** [comment]
@mgcrea 

> Looks like multiline could be a problem.

You expected multiline ('\n' support) on the markdown result in the table ? Did you try mk multiline rendering before ? I mean the only way I made it working is by remplacing the '\n' with '`<br>`' no ?

### Example :

##### Current mess :

> ## Usage
> 
> | Param | Type | Details |
> | --- | --- | --- |
> | animation _(optional)_ | `string` | Class to be added for animation purposes |
> 
>   Defaults value is `'am-fade'`.
> prefixClass _(optional)_ |  `string` | Prefix prepended to class names and events
>   Defaults value is `'alert'`.
> placement _(optional)_ |  `string` | How to position the alert
>   Defaults value is `''`.

##### HTML table version (with `<br>` ) :

> ## Usage
> 
> <table>
>   <thead>
>     <tr>
>       <th>Param</th>
>       <th>Type</th>
>       <th>Details</th>
>     </tr>
>   </thead>
>   <tbody>
>      <tr>
>       <td>
>         animation <i>(optional)</i>
>       </td>
>       <td>
>         <code>string</code>
>       </td>
>       <td>
>         <p>Class to be added for animation purposes<br>Defaults value is `'am-fade'`.</p>
>       </td>
>     </tr>
>     <tr>
>       <td>
>         prefixClass <i>(optional)</i>
>       </td>
>       <td>
>         <code>string</code>
>       </td>
>       <td>
>         <p>Prefix prepended to class names and events<br>Defaults value is `'alert'`.</p>
>       </td>
>     </tr>
>     <tr>
>       <td>
>         placement <i>(optional)</i>
>       </td>
>       <td>
>         <code>string</code>
>       </td>
>       <td>
>         <p>How to position the alert<br>Defaults value is `''`.</p>
>       </td>
>     </tr>
>   </tbody>
> </table>

**Comment by mgcrea** [comment]
@douglasduteil, yup let's go with `<br>`. Patched in 9e3bde50f0c508a09e05aadbad5b3b7db59e58af

**Comment by douglasduteil** [comment]
:+1: On peut considérer que le  `gulp readme` est fini @mgcrea ?
